docs/cluster-relocation.md

```diff
@@ -0,0 +1,82 @@
+# Cluster Relocation
+
+Hive supports the ability to migrate a `ClusterDeployment` and it's related custom resources to another Hive cluster through the use of the `ClusterRelocate` custom resource definition.
+
+## Usage
+
+In your source Hive cluster, create a Secret containing a kubeconfig for the destination Hive cluster:
+
+```bash
+$ kubectl create secret generic hub2-migration-kubeconfig --from-file=kubeconfig=./hub2.kubeconfig
+```
+
+In your source Hive cluster, create a `ClusterRelocate` custom resource: 
+
+```yaml
+apiVersion: hive.openshift.io/v1
+kind: ClusterRelocate
+metadata:
+  name: migrator
+spec:
+  kubeconfigSecretRef:
+    namespace: hive
+    name: hub2-migration-kubeconfig
+  clusterDeploymentSelector:
+    matchLabels:
+      migrateme: hub2
+```
+
+Apply the desired label to your `ClusterDeployment`:
+
+```bash 
+$ kubectl label clusterdeployment mycluster migrateme=hub2
+```
+
+The `ClusterDeployment` should appear in the destination hive, and be deleted in the source Hive, without triggering any cleanup of cluster resources.
+
+## Caveats
+
+The relocation process will migrate most of the relevant resources in a source namespace, so if you have multiple `ClusterDeployments` in one namespace, it is possible some of their secrets will be copied to the destination cluster even if only one of the `ClusterDeployments` matched the label selector. Best practice for Hive is to use a namespace per `ClusterDeployment`.
+
+## Implementation Details
+
+[Original Pull Request](https://github.com/openshift/hive/pull/1011)
+
+Relocation was implemented via a `hive.openshift.io/relocate` annotation which is added to the `ClusterDeployment`. This annotation's value can be "outgoing", "complete", or "incoming". When set, most hive controllers are disabled so as not to make changes during the relocation.
```

Comment:
This is not quite accurate. The annotation value is actually "{relocate_name}/{relocate_status}", where "{relocate_name}" is the name of the `ClusterRelocate` resource and "{relocate_status}" is one of "outgoing", "complete", or "incoming". The "{relocate_name}" is important for aborting an on-going relocation when labels or selectors change.